src/components/VectorInput/index.js

```diff
@@ -174,6 +174,19 @@ class VectorInput extends Element {
     }
 
     set value(value) {
+        if (typeof value === 'string') {
+            try {
+                // try to parse the string
+                value = JSON.parse(value);
+            } catch (e) {
+                value = [];
+            }
+            // if the string could be converted to an array but some of it's values aren't numbers
+            // then use a default array
+            if (Array.isArray(value) && value.some(i => !Number.isFinite(i))) {
```

Comment:
move this inside the try?

Reply:
I wanted to be clear which line can cause the exception. Is there a code style for this?

Reply:
You could move it inside the try and throw an exception if Number.isFinite returns true

Reply:
Done